lib/nahmii-provider.spec.js

```diff
@@ -174,6 +174,16 @@ describe('Nahmii Provider', () => {
             return provider.effectuatePayment(expectedReceipt);
         });
 
+        it('can watch a transaction hash until it is mined', async () => {
```

Comment:
@morfj curious to know how you would handle testing the polling functionality here. eg
- setting up `provider.getTransactionReceipt` to fail 3 times and succeed the 4th time
- ensuring that the promiose times out correctly